app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java

```diff
@@ -24,65 +27,81 @@
-public class SubscriberDetailFragment extends Fragment implements SubscriberDetailView {
+public class SubscriberDetailFragment extends Fragment implements SubscriberDetailView, RepositoryAdapter.ItemRepoClickListener, View.OnClickListener {
 
-    private ImageView imgAvatar;
-    private TextView txtFullName;
-    private TextView txtUserName;
-    private TextView txtCompany;
-    private TextView txtLocation;
-    private TextView txtFollowingCounter;
-    private TextView txtReposCounter;
+    private static final String ARG_USERNAME = "ARG_USERNAME";
+    private String userNameArg;
+
+    private ImageView profileImage;
+    private TextView profileFullName;
+    private TextView profileUserName;
+    private TextView profileCompany;
+    private TextView profileLocation;
+    private TextView profileFollowersCounter;
+    private TextView profileFollowingCounter;
+    private TextView profileReposCounter;
+    private String profileHtmlUrl;
+    private RepositoryAdapter repositoriesAdapter;
 
     private SubscriberDetailPresenter presenter;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        loadArgs();
+        repositoriesAdapter = new RepositoryAdapter();
+    }
+
+    public static SubscriberDetailFragment newInstance(String param1) {
+        SubscriberDetailFragment fragment = new SubscriberDetailFragment();
+        Bundle args = new Bundle();
+        args.putString(ARG_USERNAME, param1);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    private void loadArgs() {
+        if (getArguments() != null) {
+            userNameArg = getArguments().getString(ARG_USERNAME);
+        }
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View viewFragment = inflater.inflate(R.layout.fragment_subscribers_detail, container, false);
         presenter = new SubscriberDetailPresenterImpl(this);
-        /*TODO
-            The next line is a hardcoded for testing purposes
-         */
-        presenter.getUser("keyboardsurfer");
-        imgAvatar = (ImageView) viewFragment.findViewById(R.id.img_avatar_subscriber_detail);
-        txtFullName = (TextView) viewFragment.findViewById(R.id.txt_full_name_subscriber_detail);
-        txtUserName = (TextView) viewFragment.findViewById(R.id.txt_user_name_subscriber_detail);
-        txtCompany = (TextView) viewFragment.findViewById(R.id.txt_company_subscriber_detail);
-        txtLocation = (TextView) viewFragment.findViewById(R.id.txt_location_subscriber_detail);
-        txtFollowingCounter = (TextView) viewFragment.findViewById(R.id.txt_following_counter_subscriber_detail);
-        txtReposCounter = (TextView) viewFragment.findViewById(R.id.txt_repos_counter_subscriber_detail);
+
+        profileImage = (ImageView) viewFragment.findViewById(R.id.img_avatar_subscriber_detail);
```

Comment:
We think this could be improved with a dataBinding system. For example, **DataBinding** Library of Android or **butterknife**